BUG: Adjust the FieldOfView to match the aspect ratio of the slice dimensions.

Punzo opened the pull request:
This is a small fix introduced in https://github.com/Slicer/Slicer/pull/8141

Kyle asked if it can be integrated in a separate PR:
https://github.com/Slicer/Slicer/pull/8141#discussion_r2038099444

to reproduce:
1) enter in dual monitor layout
2) load example nrrd file
3) Copy and paste in the python console
```
a = slicer.mrmlScene.GetNodeByID("vtkMRMLSliceNodeRed")
b = slicer.mrmlScene.GetNodeByID("vtkMRMLSliceNodeRed+")
a.Copy(b)
```
4) you get
![image](https://github.com/user-attachments/assets/7b9d552e-ebda-4373-86ee-3d9e4fa2361a)

With this fix:
![image](https://github.com/user-attachments/assets/0a11b45e-1cff-40a2-a281-dcb8889eff8c)

@Sunderlandkyl @lassoan

lassoan commented on Libs/MRML/Core/vtkMRMLSliceNode.cxx:
```diff
@@ -951,7 +951,22 @@ void vtkMRMLSliceNode::ReadXMLAttributes(const char** atts)
   }
+
+  // Read Dimensions only if the current Dimensions are the default ones
+  if (this->Dimensions[0] == 256 && this->Dimensions[1] == 256 && this->Dimensions[2] == 1)
```
Why do we read/copy only when default value? Do we really ever want to read this from file?

(256,256,1) is a valid value, so if we want to make decision based on it then we would need to change the default to some invalid value (0,0,0) or maybe (-1,-1,-1).

Punzo replied:
Done in https://github.com/Slicer/Slicer/pull/8370/commits/52914830db2a06573ec1446dca932bb51ba8e9c9

Sunderlandkyl commented:
Thanks @Punzo, this change works well for slice sequences!

Punzo commented:
> Thanks @Punzo, this change works well for slice sequences!

thanks for testing @Sunderlandkyl. I added the MappedInLayout fix as well https://github.com/Slicer/Slicer/pull/8141#discussion_r2039878428

Can you re-test/review? thanks!

@lassoan thanks for reviewing, I have addressed your comment.

lassoan commented on Libs/MRML/Core/vtkMRMLAbstractViewNode.h:
```diff
@@ -358,6 +360,10 @@ class VTK_MRML_EXPORT vtkMRMLAbstractViewNode
 
+  ///
+  /// Define if the view is mapped in the current layout with a Qwidget.
+  int MappedInLayout;
```
Need to initialize, it can be done here or in the constructor.

Punzo replied:
a nice, I forgot, thanks@

Sunderlandkyl commented:
@Punzo Thanks, it works great!

Punzo commented on Libs/MRML/Core/vtkMRMLAbstractViewNode.h:
```diff
@@ -132,8 +132,10 @@ class VTK_MRML_EXPORT vtkMRMLAbstractViewNode
   /// \sa vtkMRMLLayoutNode::SetViewArrangement()
+  vtkSetMacro(MappedInLayout, int);
+  vtkGetMacro(MappedInLayout, int);
+  /// Deprecated. Use GetMappedInLayout instead.
```
No special reason, before an additional method was necessary because the varibale was saved in the node attributes. Since now it is a variable in the node, we can simply sue the get method. For me there is no difference, I will remove the depracation warning

Punzo commented on Libs/MRML/Core/vtkMRMLAbstractViewNode.h:
```diff
@@ -132,8 +132,10 @@ class VTK_MRML_EXPORT vtkMRMLAbstractViewNode
   /// \sa GetVisibility()
   /// \sa IsViewVisibleInLayout()
   /// \sa vtkMRMLLayoutNode::SetViewArrangement()
+  vtkSetMacro(MappedInLayout, int);
```
yes you need it in qMRMLLayoutViewFactory::setupView.

Punzo commented on Libs/MRML/Core/vtkMRMLAbstractViewNode.h:
```diff
@@ -358,6 +360,10 @@ class VTK_MRML_EXPORT vtkMRMLAbstractViewNode
 
+  ///
+  /// Define if the view is mapped in the current layout with a Qwidget.
+  int MappedInLayout;
```
I declared int because the original `IsMappedInLayout` was returning a int. But i agree it should be a bool and `IsMappedInLayout` shoudl return a bool

Punzo commented:
> @Punzo Thanks, it works great!

nice!

Punzo commented:
@lassoan thanks for reviewing the second commit. I have applied your comments.

Punzo commented:
@Sunderlandkyl @lassoan can you please have a look at this again, and in case merge? I would like to avoid conflicts with https://github.com/Slicer/Slicer/pull/8141

lassoan commented on Libs/MRML/Core/vtkMRMLAbstractViewNode.cxx:
```diff
@@ -101,6 +103,9 @@ void vtkMRMLAbstractViewNode::WriteXML(ostream& of, int nIndent)
   // value that is in the scene but what the user has set in application settings.
   // vtkMRMLWriteXMLFloatMacro(screenScaleFactor, ScreenScaleFactor);
 
+  // Do not write MappedInLayout attribute, as it is not a property of the view node.
```
```suggestion
  // Do not write MappedInLayout attribute, as it is set dynamically by the layout manager. It should not be copied between objects or saved to file.
```

Sunderlandkyl commented:
@Punzo Changes look good to me! @lassoan any other comments?

lassoan commented on Libs/MRML/Core/vtkMRMLAbstractViewNode.h:
```diff
@@ -358,6 +358,10 @@ class VTK_MRML_EXPORT vtkMRMLAbstractViewNode
   vtkSmartPointer<vtkStringArray> AxisLabels;
 
+  ///
+  /// Define if the view is mapped in the current layout with a Qwidget.
```
MRML is not aware of Qt.

```suggestion
  /// Define if the view is displayed in the current layout.
```

lassoan reviewed (approved):
Looks good to me, thank you.
